src/include/miopen/logger.hpp

```diff
@@ -300,7 +318,14 @@ std::ostream& LogParam(std::ostream& os, std::string name, const std::vector<T>&
             miopen_log_func_ss << miopen::LoggingPrefix() << "}" << std::endl;          \
             std::cerr << miopen_log_func_ss.str();                                      \
         }                                                                               \
-    while(false)
+        if(miopen::IsLoggingToRoctx())                                                  \
+        {                                                                               \
+            std::ostringstream miopen_log_func_ss;                                      \
+            miopen_log_func_ss << "s_api = "<< __FUNCTION__ << " | ";                   \
+            MIOPEN_PP_EACH_ARGS(MIOPEN_LOG_FUNCTION_EACH_ROCTX, __VA_ARGS__)            \
+            logtx.logRange(miopen_log_func_ss.str().c_str());                           \
```

Comment:
`logRange` gets `std::string` as parameter and calls `c_str` internally. In this case it constructs `std::string` just for no reason.

```suggestion
            logtx.logRange(miopen_log_func_ss.str());                                   \
```

Reply:
Opps, I think I changed the arg type there.  Good catch.